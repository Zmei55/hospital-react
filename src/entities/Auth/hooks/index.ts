export * from "./useLogIn";
export * from "./useLogOut";
export * from "./useHandleChange";
