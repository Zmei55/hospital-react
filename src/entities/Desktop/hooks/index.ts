export * from "./useNewRequest";
export * from "./useFindRequest";
