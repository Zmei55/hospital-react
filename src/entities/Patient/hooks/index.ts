export * from "./useChoosePatient";
export * from "./useGetPatient";
export * from "./usePatientsListFind";
export * from "./useHandleInfoActive";
export * from "./useClearPatient";
export * from "./useTogglePatientsModal";
