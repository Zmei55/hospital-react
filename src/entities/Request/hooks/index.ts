export * from "./useRequestServices";
export * from "./useSendRequestDetails";
