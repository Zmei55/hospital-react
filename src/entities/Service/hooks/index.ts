export * from "./useServiceAdd";
export * from "./useServiceDelete";
export * from "./useSelectedServiceDelete";
export * from "./useSelectedServiceListSave";
export * from "./useSelectedServiceListClear";
export * from "./useGetLaborsList";
export * from "./useGetServiceById";
export * from "./useGetServiceListById";
export * from "./useHandleLaborChange";
export * from "./useHandleServicesForm";
export * from "./useToggleLaborModal";
export * from "./useToggleServicesModal";
