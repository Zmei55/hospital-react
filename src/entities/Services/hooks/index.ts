export * from "./useSavedServicesList";
export * from "./useServicesFilter";
export * from "./useAddService";
