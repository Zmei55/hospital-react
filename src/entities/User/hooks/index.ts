export * from "./useGetUserData";
export * from "./useToggleMenuModal";
