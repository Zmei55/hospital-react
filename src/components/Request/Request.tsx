import { useAppDispatch } from 'types';
import { fetchPatient } from 'redux/appSlice';
import { Patient } from 'components/Patient';
import { Services } from 'components/Services';
import { Containers } from 'components/Containers';

import {
  Container,
  Header,
  Body,
  RouterLink,
  HeaderTitle,
  CloseBtn,
  Box,
  IconCross,
} from './Request.styled';

type TPatient = {
  _id: string;
  name: string;
  birthDate: string;
  cardNumber: number;
  gender: string;
  phoneNumber: string;
  email: string;
  identityDocument: string;
  residenceAddress: {
    street: string;
    houseNumber: string;
    city: string;
    postcode: string;
  };
  services: never[];
  containers: never[];
};

type TInitialState = {
  data: {
    result: TPatient;
  };
};

const initialPatient: TInitialState = {
  data: {
    result: {
      _id: '',
      name: '',
      birthDate: '',
      cardNumber: 0,
      gender: '',
      phoneNumber: '',
      email: '',
      identityDocument: '',
      residenceAddress: {
        street: '',
        houseNumber: '',
        city: '',
        postcode: '',
      },
      services: [],
      containers: [],
    },
  },
};

export const Request: React.FC = () => {
  const dispatch = useAppDispatch();

  const clearPatient = (): void => {
    dispatch(fetchPatient(initialPatient));
  };

  return (
    <Container>
      <Header>
        <HeaderTitle>Neue Bestellung</HeaderTitle>
        <RouterLink to="/desktop">
          <CloseBtn type="button" onClick={clearPatient}>
            <IconCross />
          </CloseBtn>
        </RouterLink>
      </Header>

      <Body>
        <Patient />

        <Box>
          <Services />

          <Containers />
        </Box>
      </Body>
    </Container>
  );
};
