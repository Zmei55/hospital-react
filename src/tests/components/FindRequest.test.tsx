import { render, screen, cleanup } from "@testing-library/react";
import { allTheProviders, requestListMocks } from "tests";
import * as sharedHooks from "shared/hooks";
import * as requestHooks from "entities/Request/hooks";
import { FindRequest } from "entities/Request";

afterEach(() => {
  cleanup();
});

jest.mock("shared/hooks");
jest.mock("entities/Request/hooks");

const mockedUseNavigate = jest.spyOn(sharedHooks, "useAppNavigate");
const mockedUseFilterRequest = jest.spyOn(requestHooks, "useFilterRequest");
const mockedUseChooseRequest = jest.spyOn(requestHooks, "useChooseRequest");

describe("FindRequest component", () => {
  beforeEach(() => {
    mockedUseNavigate.mockReturnValue({ handleNavigate: jest.fn() });
    mockedUseChooseRequest.mockReturnValue({ chooseRequest: jest.fn() });
  });

  it("FindRequest renders", () => {
    mockedUseFilterRequest.mockReturnValue({
      requestList: null,
      setRequestList: jest.fn(),
      handleFilterRequests: jest.fn(),
      isLoading: false,
      isError: false,
    });

    render(allTheProviders(<FindRequest />));

    expect(screen.getByTestId("find-request-container")).toBeInTheDocument();
  });

  it("FindRequest filter render", () => {
    mockedUseFilterRequest.mockReturnValue({
      requestList: null,
      setRequestList: jest.fn(),
      handleFilterRequests: jest.fn(),
      isLoading: false,
      isError: false,
    });

    render(allTheProviders(<FindRequest />));

    expect(screen.getByTestId("find-request-header")).toBeInTheDocument();
    expect(screen.getByTestId("find-request-header-title")).toBeInTheDocument();
    expect(screen.getByText("Find a request")).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /reset/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /find/i })).toBeInTheDocument();
    expect(screen.getByTestId("close-find-request-btn")).toBeInTheDocument();
    expect(screen.getByTestId("find-request-body")).toBeInTheDocument();
    expect(screen.getByTestId("find-request-form")).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText(/first name and last name/i)
    ).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/card number/i)).toBeInTheDocument();
    expect(
      screen.getByTestId("find-request-body-request-number-input")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("find-request-body-request-date-input")
    ).toBeInTheDocument();
  });

  it("FindRequest requestList renders", () => {
    mockedUseFilterRequest.mockReturnValue({
      requestList: requestListMocks,
      setRequestList: jest.fn(),
      handleFilterRequests: jest.fn(),
      isLoading: false,
      isError: false,
    });

    const { container } = render(allTheProviders(<FindRequest />));

    expect(screen.getByTestId("find-request-list")).toBeInTheDocument();
    expect(screen.getByTestId("find-request-list-header")).toBeInTheDocument();
    expect(screen.getByText("First name and last name")).toBeInTheDocument();
    expect(screen.getByText("Card number")).toBeInTheDocument();
    expect(screen.getByText("Request number")).toBeInTheDocument();
    expect(screen.getByText("Request date")).toBeInTheDocument();
    expect(screen.getByTestId("find-request-list-body")).toBeInTheDocument();

    const listItem = screen.getAllByTestId("find-request-listitem")[0];
    expect(listItem).toBeInTheDocument();

    const patientName = screen.getAllByTestId("find-request-listitem-name")[0];
    expect(patientName).toBeInTheDocument();

    const cardNumber = screen.getAllByTestId(
      "find-request-listitem-card-number"
    )[0];
    expect(cardNumber).toBeInTheDocument();

    const requestNumber = screen.getAllByTestId(
      "find-request-listitem-request-number"
    )[0];
    expect(requestNumber).toBeInTheDocument();

    const requestDate = screen.getAllByTestId(
      "find-request-listitem-request-date"
    )[0];
    expect(requestDate).toBeInTheDocument();

    const selectBtn = screen.getAllByTestId("select-found-request")[0];
    expect(selectBtn).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
