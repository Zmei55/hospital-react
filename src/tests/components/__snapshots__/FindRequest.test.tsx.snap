// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindRequest component FindRequest requestList renders 1`] = `
<div>
  <div
    class="css-1hul4ku"
    data-testid="find-request-container"
  >
    <div
      class="css-ixaifj"
      data-testid="find-request-header"
    >
      <h2
        class="css-w70xkr"
        data-testid="find-request-header-title"
      >
        Find a request
      </h2>
      <button
        class="css-ch4pr2"
        data-testid="reset-find-request-btn"
        form="findReqForm"
        style="height: 72px; padding-right: 44px; padding-left: 44px; margin-right: 24px;"
        type="button"
      >
        Reset
      </button>
      <button
        class="css-mbrcjd"
        data-testid="find-find-request-btn"
        form="find-request-form"
        style="height: 72px; padding-right: 44px; padding-left: 44px; margin-right: 24px;"
        type="submit"
      >
        Find
      </button>
      <button
        class="css-14jjab4"
        data-testid="close-find-request-btn"
        style="width: 72px; height: 72px;"
      >
        <svg
          class="css-bvj43j"
          fill="white"
          height="48"
          width="48"
        >
          <use
            href="svgSprite.svg#cross"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-r5mrjc"
      data-testid="find-request-body"
    >
      <form
        class="css-156aa7i"
        data-testid="find-request-form"
        id="find-request-form"
      >
        <label
          class="css-gz4v1g"
        >
          <input
            class="css-1ry88mf"
            data-testid="find-request-body-name-input"
            name="patientName"
            placeholder="First name and last name"
            style="width: 100%;"
          />
        </label>
        <label
          class="css-gz4v1g"
        >
          <input
            class="css-1ry88mf"
            data-testid="find-request-body-card-number-input"
            name="cardNumber"
            placeholder="Card number"
            style="width: 100%;"
          />
        </label>
        <label
          class="css-gz4v1g"
        >
          <input
            class="css-1ry88mf"
            data-testid="find-request-body-request-number-input"
            name="requestNumber"
            placeholder="Request number"
            style="width: 100%;"
          />
        </label>
        <label
          class="css-gz4v1g"
        >
          <input
            class="css-1ry88mf"
            data-testid="find-request-body-request-date-input"
            name="createdAt"
            placeholder="Request date"
            style="width: 100%;"
            type="date"
          />
        </label>
      </form>
      <div
        class="css-1locwai"
        data-testid="find-request-list"
      >
        <div
          class="css-1coqfd9"
          data-testid="find-request-list-header"
        >
          <div
            class="css-9dis78"
          >
            First name and last name
          </div>
          <div
            class="css-1uhsm05"
          >
            Card number
          </div>
          <div
            class="css-1uhsm05"
          >
            Request number
          </div>
          <div
            class="css-16sqn77"
          >
            Request date
          </div>
          <div
            class="css-1q4b8hs"
          />
        </div>
        <ul
          class="css-yquj6e"
          data-testid="find-request-list-body"
        >
          <li
            class="css-hwfa6i"
            data-testid="find-request-listitem"
          >
            <p
              class="css-jc3204"
              data-testid="find-request-listitem-name"
            >
              Emma Weber
            </p>
            <p
              class="css-1sh92no"
              data-testid="find-request-listitem-card-number"
            >
              123456789
            </p>
            <p
              class="css-1sh92no"
              data-testid="find-request-listitem-request-number"
            >
              1
            </p>
            <p
              class="css-xjyctd"
              data-testid="find-request-listitem-request-date"
            >
              
  08
  .05
  .2023
            </p>
            <p
              class="css-17ol73b"
              data-testid="find-request-listitem-select-btn"
            >
              <button
                class="css-1x5y6pv"
                data-testid="select-found-request"
              >
                Select
              </button>
            </p>
          </li>
          <li
            class="css-hwfa6i"
            data-testid="find-request-listitem"
          >
            <p
              class="css-jc3204"
              data-testid="find-request-listitem-name"
            >
              Emma Weber
            </p>
            <p
              class="css-1sh92no"
              data-testid="find-request-listitem-card-number"
            >
              123456789
            </p>
            <p
              class="css-1sh92no"
              data-testid="find-request-listitem-request-number"
            >
              2
            </p>
            <p
              class="css-xjyctd"
              data-testid="find-request-listitem-request-date"
            >
              
  08
  .05
  .2023
            </p>
            <p
              class="css-17ol73b"
              data-testid="find-request-listitem-select-btn"
            >
              <button
                class="css-1x5y6pv"
                data-testid="select-found-request"
              >
                Select
              </button>
            </p>
          </li>
          <li
            class="css-hwfa6i"
            data-testid="find-request-listitem"
          >
            <p
              class="css-jc3204"
              data-testid="find-request-listitem-name"
            >
              Emma Weber
            </p>
            <p
              class="css-1sh92no"
              data-testid="find-request-listitem-card-number"
            >
              123456789
            </p>
            <p
              class="css-1sh92no"
              data-testid="find-request-listitem-request-number"
            >
              7
            </p>
            <p
              class="css-xjyctd"
              data-testid="find-request-listitem-request-date"
            >
              
  12
  .05
  .2023
            </p>
            <p
              class="css-17ol73b"
              data-testid="find-request-listitem-select-btn"
            >
              <button
                class="css-1x5y6pv"
                data-testid="select-found-request"
              >
                Select
              </button>
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
